src/flepimop2/_utils/_click.py

```diff
@@ -0,0 +1,47 @@
+from typing import TypeVar
+
+from click import BadOptionUsage
+
+T = TypeVar("T")
+U = TypeVar("U")
+
+
+def _override_or_val(override: T | None, value: U) -> T | U:
+    """
+    Return the override value if it is not None, otherwise return the value.
+
+    Args:
+        override: Optional override value.
+        value: The value to return if the override is None.
+
+    Returns:
+        The `override` value if it is not None, otherwise `value`.
+    """
+    return value if override is None else override
+
+
+def _get_config_target(group: dict[str, T], name: str | None) -> T:
+    """
+    Get a `T` by name from a group.
+
+    Args:
+        name: The name of the module target. If `None`, defaults to the first item
+          in group.
+        group: The module group to get the target from.
+
+    Returns:
+        The `T` with the specified name.
+
+    Raises:
+        click.BadOptionUsage: If the specified name is not found in the group.
+    """
+    if name is None:
+        name = next(iter(group.keys()))
+    res = group.get(name)
+    if res is None:
+        msg = f"""
+        Module target '{name}' not found in configuration.
+        Available targets: {group.keys()}
+        """
+        raise BadOptionUsage(option_name="target", message=msg)
+    return res
```

Comment:
1. There's a potential sharp edge here if `group` is empty. This will cause `next(iter(group.keys()))` to raise a `StopIteration`. There are two options: 1) Add an `if not group:` at the beginning and raise an error, or 2) change the line to `next(iter({}.keys()), None)` so you get `None` for `name` and catch it with the `BadOptionUsage` error.
2. Can `msg` be a single line rather than multiline string? Multiline strings, especially with tabs, look a bit weird for short error messages.